crates/pixi_command_dispatcher/src/source_build/mod.rs

```diff
@@ -102,102 +111,271 @@ impl SourceBuildSpec {
+            })
+            .await
+            .map_err_with(Box::new)
+            .map_err_with(SourceBuildError::InstallBuildEnvironment)?;
 
-        Ok(BuiltSource {
-            source: source_checkout,
-            input_globs: built_package.input_globs,
-            output_file: built_package.output_file,
-        })
+        // Install the host environment.
+        let _host_prefix = command_dispatcher
+            .install_pixi_environment(InstallPixiEnvironmentSpec {
+                name: format!("{} (host)", self.source.package_record.name.as_source()),
+                records: host_records,
+                prefix: Prefix::create(&directories.host_prefix)
+                    .map_err(SourceBuildError::CreateBuildEnvironmentDirectory)
+                    .map_err(CommandDispatcherError::Failed)?,
+                installed: None,
+                build_environment: self.build_environment.to_build_from_build(),
+                force_reinstall: Default::default(),
+                channels: self.channels.clone(),
+                channel_config: self.channel_config.clone(),
+                variants: self.variants.clone(),
+                enabled_protocols: self.enabled_protocols.clone(),
+            })
+            .await
+            .map_err_with(Box::new)
+            .map_err_with(SourceBuildError::InstallBuildEnvironment)?;
+
+        command_dispatcher
+            .backend_source_build(BackendSourceBuildSpec {
+                backend,
+                record: self.source,
+                method: BackendSourceBuildMethod::BuildV2(BackendSourceBuildV2Method {
+                    build_prefix: BackendSourceBuildPrefix {
+                        platform: self.build_environment.build_platform,
+                        prefix: directories.build_prefix,
+                    },
+                    host_prefix: BackendSourceBuildPrefix {
+                        platform: self.build_environment.host_platform,
+                        prefix: directories.host_prefix,
+                    },
+                    variant: output.metadata.variant,
+                    output_directory: self.output_directory,
+                }),
+            })
+            .await
+            .map_err_with(SourceBuildError::from)
+    }
+
+    async fn solve_dependencies(
+        &self,
+        name: String,
+        command_dispatcher: &CommandDispatcher,
+        dependencies: Dependencies,
+        build_environment: BuildEnvironment,
+    ) -> Result<Vec<PixiRecord>, CommandDispatcherError<SolvePixiEnvironmentError>> {
+        if dependencies.dependencies.is_empty() {
+            return Ok(vec![]);
+        }
+        command_dispatcher
+            .solve_pixi_environment(PixiEnvironmentSpec {
+                name: Some(name),
+                dependencies: dependencies.dependencies,
+                constraints: dependencies.constraints,
+                installed: vec![], // TODO: To lock build environments, fill this.
```

Comment:
Sounds important.

Reply:
We currently do not yet lock the host and or build environments. There remains some tricky edge cases that we need to think about. We can add this later I think.